sol/Usdx.sol

```diff
@@ -39,12 +39,8 @@ contract USDX is ERC20, Ownable {
 	}
 
 	receive() external payable {
-		uint256 decs = ethUsdPriceFeed.decimals();
-		// 10**64 < 2**213; safely stored in uint256.
-		require(decs <= 64, "feed decimals too large");
-
 		(,int256 rate,,,) = ethUsdPriceFeed.latestRoundData();
-		uint256 usd = msg.value.mul(rate.toUint256()).div(10**decs);
+		uint256 usd = msg.value.mul(rate.toUint256()).div(10**8); //latestRoundData uses 8 decimals
```

Comment:
For a bit more added safety, thoughts on creating:
```
uint8 constant FEED_DECS = 8;
```
And then in `setFeed`, we do something like:
```
	// _newFeed's decimals must be 8 and immutable
	function setFeed(address _newFeed) public onlyOwner {
		AggregatorV3Interface newFeed = AggregatorV3Interface(_newFeed);
		requre(newFeed.decimals() == FEED_DECS);
		ethUsdPriceFeed = newFeed;
	}
```
This way, we can never accidentally shoot ourselves in the foot (and we can specifically call out the requirements/assumptions made by the priceFeed).

Reply:
Been looking into the heartbeat now...I'll update you later if I figure more of it out (the docs don't seem to mention too much unfortunately). I think that would be a great idea though - in implementation I'm thinking maybe we could use the answeredInRound param and compare it with the current roundId at the end of the heartbeat (so current time = start + 3 hours) to see if the answer was updated in the round. 

With adding FEED_DECS and altering setFeed, that definitely looks like a safer approach. I'll update that in a moment

Reply:
Just pushed the updated usdx.sol/test cases, let me know if anything should be fixed

Reply:
Change `div(10**8)` to use `FEED_DECS`, and it LGTM.  Feel free to merge after that change.

Regarding heartbeat...I don't see any "heartbeat" related methods in the chainlink `AggregatorV3Interface.sol` interface we're using, so not sure how exactly this is supposed to be used.  Might be worth asking in the chainlink discord to see if this is still used.

Reply:
Yeah, I don't think it's used directly in that class at all. I found its usage in `FluxAggregator.sol` though which I think `AggregatorV3Interface.sol` uses. Sent you a message on discord - I think I've made some sense of a way we could use it but not entirely sure. 

And sounds good! I'll merge in a bit